### PR TITLE
C++appln

### PR DESCRIPTION
**Pull request by Abdulkadir-eInfo**
Updated C++ Application Code;
-Implemented STL,
-Autoincrement ID,
-Function Name(Camelcase)

### CODE REVIEW

File: MyStudentSystem.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            FIND=2,
+            FIND2=3,
+            DELETE=4,
+            UPDATE=5,
+            DISPLAYALL=6,
+            DISPLAYALL2=7,
+            QUIT=8
+    };
+
+    while(8!=choice) //!< Loop until user enters 8(Choice for Quitting out)
```
Comment (girishbvyas1982) [inline MyStudentSystem.cpp]:
```suggestion
    while(QUIT!=choice) //!< Loop until user enters 8(Choice for Quitting out)
```

File: include/Base.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#ifndef BASE_H
+#define BASE_H
+#include <vector>
+
+
+class BaseOperations
+{
+	public:
+	int ID=100;
```
Comment (girishbvyas1982) [inline include/Base.h]: What is 100 ? should be defined as const expressiong

File: include/Base.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	struct mystruct //!< Structure of all the student details
+	{
+		int id;
+		std::string name;
+    	std::string branch;
```
Comment (girishbvyas1982) [inline include/Base.h]: correct indentation

File: include/Base.h (hunk 2 above)
Comment (girishbvyas1982) [inline include/Base.h]: Why this is public?

File: include/Base.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    	std::string location;
+	}st;
```
Comment (girishbvyas1982) [inline include/Base.h:16]: why this is public?

File: include/Base.h [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+	std::vector<mystruct> vect; //!< Vector to datatype mystruct that holds all the student detail
```
Comment (girishbvyas1982) [inline include/Base.h:18]: why this is public ?

File: include/Base.h [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+	
+	BaseOperations(); //!< Constructor that transfers data from txt file into the vector
+	void EnterDetail(); //!< Function is doing nothing in the parent class, will be overrided in the child class
+	int CheckName(std::string); //!< Checks whether given Name is already present in the list
```
Comment (girishbvyas1982) [inline include/Base.h]:
```suggestion
	int CheckName(cost std::string &); //!< Checks whether given Name is already present in the list
```

Comment (girishbvyas1982) [reply include/Base.h]: please understand this and answer why it is required.

File: include/Base.h [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+	virtual int ShowDetail(std::string)=0; //!< Showdetail function has no implementation here, will be implemented in dervied class
```
Comment (girishbvyas1982) [inline include/Base.h]: same comment

File: src/Base.cpp [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+/** Checks whether given Name is already present in the list */
+int BaseOperations::CheckName(string checkName)
+{
+    vector<mystruct>::iterator ptr;
+    ptr=vect.begin();
+    mystruct find_ptr;
+    for (ptr = vect.begin(); ptr < vect.end(); ptr++)
+    {
+        mystruct mptr=*ptr;
+        if(mptr.name==checkName){
```
Comment (girishbvyas1982) [inline src/Base.cpp]:
```suggestion
        if(mptr.name==checkName)
        {
```

File: src/Base.cpp [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+            return -1;
```
Comment (girishbvyas1982) [inline src/Base.cpp]: use enum success of fail instead

File: src/DisplayOperations.cpp [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+#include <fstream>
+#include <exception>
+#include "DisplayOperations.h"
+using namespace std;
+
+/** Displays all the student entries */
+void DisplayAllOperation::DisplayAll()
+{
+    int size=vect.size();
+    if(0==size)
```
Comment (girishbvyas1982) [inline src/DisplayOperations.cpp]:
```suggestion
    if(0==vect.size())
```

File: src/DisplayOperations.cpp [hunk 11]
```diff
@@ -0,0 +1,42 @@
+#include <iostream>
+#include <fstream>
+#include <exception>
+#include "DisplayOperations.h"
+using namespace std;
+
+/** Displays all the student entries */
+void DisplayAllOperation::DisplayAll()
+{
+    int size=vect.size();
```
Comment (girishbvyas1982) [inline src/DisplayOperations.cpp]:
```suggestion
```

File: src/DisplayOperations.cpp [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        cout << "No Student Record Present" << endl;
+    }
+    else
+    {
+        cout << "\nId\tName\tBranch\tLocation" << endl;
+        vector<mystruct>::iterator ptr;
+        ptr=vect.begin();
+        mystruct mptr;
+        for (ptr = vect.begin(); ptr < vect.end(); ptr++)
```
Comment (girishbvyas1982) [inline src/DisplayOperations.cpp]: you can use range based loops ref: https://www.geeksforgeeks.org/range-based-loop-c/

File: include/Base.h (hunk 2 above)
Comment (Abdulkadir-eInfo) [reply include/Base.h:9]: Initialization of ID,ID starts with 100.

File: include/Base.h (hunk 3 above)
Comment (Abdulkadir-eInfo) [reply include/Base.h:14]: It is correclty showing in VScode, while pushing it to Git; it gets changed.

File: src/DisplayOperations.cpp [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+        {
+            mptr=*ptr;
+            if(ID<mptr.id)
```
Comment (girishbvyas1982) [inline src/DisplayOperations.cpp]: optimize it with single line statement

File: src/DisplayOperations.cpp [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+                ID=mptr.id;
+            cout << mptr.id << "\t" << mptr.name << "\t" << mptr.branch << "\t" << mptr.location  << endl;
+            
+        }
+    }
+}
+
+/** Displayall function that displays all entries along with count of entries */
+int DisplayAllOperation::DisplayAll(int x){
```
Comment (girishbvyas1982) [inline src/DisplayOperations.cpp]:
```suggestion
int DisplayAllOperation::DisplayAll(int x)
{
```

File: src/Operations.cpp [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+#include <fstream>
+#include <exception>
+#include "Operations.h"
+using namespace std;
+
+/** Enter student details in the file */
+int Operations::EnterDetail()
+{
+    int check=0;
+	cout << "\nEnter the student details:" << endl;
```
Comment (girishbvyas1982) [inline src/Operations.cpp]: correct indentation

File: src/Operations.cpp [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+	cout << "NAME:";
+	cin >> st.name;
+
+    check=CheckName(st.name);
+    if(-1==check){
```
Comment (girishbvyas1982) [inline src/Operations.cpp]: keep same style, lot of places I found mismatch

File: src/Operations.cpp [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+    int is_studentdetail_present=0;
+    
+    if(0==size)
+    {
+        cout << "No Student Record Present" << endl;
+        return -1;
+    }
+    else
+    {
+        vector<mystruct>::iterator ptr;
```
Comment (girishbvyas1982) [inline src/Operations.cpp]: optimize this logic looks lot of code can be reomved

File: src/Operations.cpp [hunk 18]
```diff
@@ -0,0 +1,197 @@
...
+    string find_name;
+    cout << "Enter the Name of Student you want to delete:" << endl;
+    cin >> find_name;
+    if(0==size)
+    {
+        cout << "No Student Record Present" << endl;
+    }
+    else
+    {
+        vector<mystruct>::iterator ptr;
```
Comment (girishbvyas1982) [inline src/Operations.cpp:103]: repeating same logic can be put in common function and call function instead

File: src/Operations.cpp [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+        }
+        file_write.close();
+    }
+}
+
+/** Updates Student Details with new details entered by User */
+void Operations::UpdateDetail(string Update_name)
+{
+    string name,branch, location;
+    int is_update_succesful=0;
```
Comment (girishbvyas1982) [inline src/Operations.cpp:152]: why we need "is_update_succesful"

File: src/Operations.cpp [hunk 20]
```diff
@@ -0,0 +1,197 @@
...
+    if(-1!=st.id)
+    {
+        cout << "Enter Updated Details:" << endl;
+        cout << "NAME:";
+        cin >> st.name;
+        cout << "Branch:";
+        cin >> st.branch;
+        cout << "Location:";
+        cin >> st.location;
+        vector<mystruct>::iterator ptr;
```
Comment (girishbvyas1982) [inline src/Operations.cpp]: optimize code to as much less lines as possible

File: include/Base.h (hunk 3 above)
Comment (girishbvyas1982) [reply include/Base.h]: check tab and single space

**Pull request closed without merging**